bot_example/src/delegate_call_in_loop.rs

```diff
@@ -0,0 +1,95 @@
+use std::collections::BTreeMap;
+use std::error::Error;
+
+use aderyn_driver::core_ast::MemberAccess;
+
+use aderyn_driver::context::{browser::ExtractMemberAccesses, loader::ContextLoader};
+use aderyn_driver::detection_modules::capture;
+use aderyn_driver::detector::{Detector, IssueSeverity};
+
+#[derive(Default)]
+pub struct DelegateCallInLoopDetector {
+    // Keys are source file name and line number
+    found_instances: BTreeMap<(String, usize), String>,
+}
+
+impl Detector for DelegateCallInLoopDetector {
+    fn detect(&mut self, loader: &ContextLoader) -> Result<bool, Box<dyn Error>> {
+        let mut member_accesses: Vec<MemberAccess> = vec![];
+
+        // Get all delegatecall member accesses inside for statements
+        member_accesses.extend(loader.for_statements.keys().flat_map(|statement| {
+            ExtractMemberAccesses::from(statement)
```

Comment:
Does the parent finder work here? `source_unit_of`, `contract_definition_of`, etc?

Reply:
Yes ! `GetParent::XX` works - just tested it! 

All the stuff from the `browser` module is visible : )